src/server_manager/install_scripts/install_server.sh

```diff
@@ -81,18 +94,59 @@ function log_for_sentry() {
 # Check to see if docker is installed.
 function verify_docker_installed() {
   if ! command_exists docker; then
-    log_error "Docker CE must be installed, please run \"curl -sS https://get.docker.com/ | sh\" or visit https://docs.docker.com/install/"
-    exit 1
+    log_error "FAILED"
+    echo -n "> Would you like to install Docker? [Y/n] "
+    if ! run_step_with_user_confirmation "Installing Docker" install_docker; then
```

Comment:
I think this would be a little more readable:
```bash
read -p "> Would you like to install Docker? [Y/n] " INSTALL_RESPONSE
if [[ ${INSTALL_RESPONSE:0:1} == "y" || ${INSTALL_RESPONSE:0:1} == "Y"]]; then
 run_step ....
fi
```
No need for the extra function.

Reply:
Without the function we would have to write that if statement every time we want a user response, whereas `run_step_with_user_confirmation` already takes care of processing the user response, delegating the prompt to the caller.

Reply:
I find it harder to read to have the prompt separate from where you read the input.
You should either move the prompt inside the function, or move the input outside the function so prompt and input are collocated.

Notice that the prompt could be simplified with a function:
```bash
if confirm "Would you like to install Docker?"; then
  run_step "Installing Docker" install_docker
fi
```
`confirm` would show the prompt and get the input. (it would also remove the prompt format duplication)
That would be quite clean, and make it clear when we actually run install_docker.